micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/resttemplate/MetricsAspectIntegrationSpec.groovy

```diff
@@ -0,0 +1,154 @@
+package com.ofg.infrastructure.web.resttemplate
+
+import com.codahale.metrics.MetricRegistry
+import com.codahale.metrics.Timer
+import com.ofg.infrastructure.base.MvcWiremockIntegrationSpec
+import com.ofg.infrastructure.web.resttemplate.custom.RestTemplate
+import com.ofg.infrastructure.web.resttemplate.fluent.ServiceRestClientConfiguration
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.SpringApplicationContextLoader
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.EnableAspectJAutoProxy
+import org.springframework.context.support.PropertySourcesPlaceholderConfigurer
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpMethod
+import org.springframework.http.RequestEntity
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.web.client.RequestCallback
+import org.springframework.web.client.ResponseExtractor
+import org.springframework.web.client.RestOperations
+
+import javax.annotation.Resource
+import java.util.concurrent.Callable
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*
+
+@ContextConfiguration(classes = [ServiceRestClientConfiguration, Config], loader = SpringApplicationContextLoader)
+class MetricsAspectIntegrationSpec extends MvcWiremockIntegrationSpec {
+
+    @Resource
+    RestOperations restTemplate
+
+    @Autowired
+    MetricRegistry metricRegistry
+
+    URI wireMockUri
+
+    def setup() {
+        wireMockUri = new URI(wireMockUrl)
+        stubInteraction(any(urlMatching('/.*')), aResponse().withStatus(200))
+    }
+
+    def 'should elide subsequent path parts in metric name'() {
+        expect:
```

Comment:
I tried really hard to do that, but then every closure would have 2 parameters. where-blocks can only access static scope and I need the restTemplate and wireMockUrl instance fields, so I'd prefer it this way. Why would you like a where-block? To have failures reported separately? Then this test would count for like 40 tests, which wouldn't be fair I think ;).